src/wallet/rpcdump.cpp

```diff
@@ -565,7 +565,7 @@ UniValue importwallet(const JSONRPCRequest& request)
         if (!file.is_open()) {
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot open wallet dump file");
         }
-        Optional<int> tip_height = locked_chain->getHeight();
+        Optional<int> tip_height = pwallet->chain().getHeight();
```

Comment:
Why use chain height instead of wallet height here and other places? Anywhere the wallet is locked and the chain isn't locked it would seem safer to use the wallet height.

Reply:
I think I can switch for the one in `CreateTransaction`, all left are tied to the rescan logic which doesn't call `BlockConnected` and so doesn't update `m_last_block_processed` accurately. I can rework a bit the rescan logic to make it work on m_last_block_processed_height but I felt it was a bit of scope and an increased burden for reviewers.

As you know I plan to rework the rescan logic after this PR. At term it would remove all `getHeight` callsites`.

Reply:
Thanks, chain height seems fine for rescan logic, and is what I'd expect there. I think the other places should use wallet height if there isn't an explicit reason not to.

Reply:
Corrected on 24f40fc, there was only one occurrence in `CreateTransaction`. All others are tied to rescan logic, including ones in `rpcdump`/`rpcwallet`.